src/main/java/com/minecolonies/coremod/commands/colonycommands/RefreshColonyCommand.java

```diff
@@ -45,15 +53,61 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         return super.getCommandUsage(sender) + "<ColonyId|OwnerName>";
     }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final List<ActionArgument> actionArgumentList,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 33 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)